### PR TITLE
paperless-ngx: 1.17.4 -> 2.0.1

### LINKED ISSUE #270952: Update request: paperless-ngx 1.17.4 → 2.0.0
- Package name: paperless-ngx
- Latest released version: `2.0.0` ([paperless-ngx/paperless-ngx 2.0.0 release](https://github.com/paperless-ngx/paperless-ngx/releases/tag/v2.0.0))

- Current version on the unstable channel: `1.17.4`
- Current version on the stable/release channel: `1.17.2` (`nixos-23.05`), `1.17.4` (`nixos-23.11`)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**: @lukegb @gador @erikarvstedt 

-----

Note for maintainers: Please tag this issue in your PR.

**Priorities**

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by leona-ya**
## Description of changes

https://github.com/paperless-ngx/paperless-ngx/releases/tag/v2.0.0
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v2.0.1

Fixes #270952

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/python-ipware/default.nix [hunk 1]
```diff
@@ -3,24 +3,30 @@
 , fetchFromGitHub
 , pythonOlder
 , unittestCheckHook
+, setuptools
 }:
 buildPythonPackage rec {
   pname = "python-ipware";
-  version = "0.9.0";
-  format = "setuptools";
+  version = "2.0.0";
```
Comment (lukegb) [inline pkgs/development/python-modules/python-ipware/default.nix:10]: I can't check from here, but are there any dependencies on python-ipware and do they still work after this bump?

**Review by lukegb — APPROVED** [review]
Seems fine. I probably wouldn't have removed the indirection via the `python` attribute on `paperless-ngx` immediately in case we need it again later.

**Comment by erikarvstedt** [comment]
I agree with @lukegb to bring back the `python` variable.

File: pkgs/development/python-modules/django-auditlog/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    setuptools
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    django
+    python-dateutil
+  ];
+
+  # requires postgresql
```
Comment (natsukium) [inline pkgs/development/python-modules/django-auditlog/default.nix:36]: Can we use `postgresqlTestHook`?

File: pkgs/development/python-modules/gotenberg-client/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+, fetchFromGitHub
+, pythonOlder
+, hatchling
+, httpx
+, typing-extensions
+}:
+buildPythonPackage rec {
+  pname = "gotenberg-client";
+  version = "0.3.0";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/gotenberg-client/default.nix:12]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/gotenberg-client/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    httpx
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    typing-extensions
+  ] ++ httpx.optional-dependencies.http2;
+
+  pythonImportsCheck = [
+    "gotenberg_client"
+  ];
+
+  doChecks = false;
```
Comment (natsukium) [inline pkgs/development/python-modules/gotenberg-client/default.nix:34]: Please comment on why we need to disable tests.

File: pkgs/development/python-modules/gotenberg-client/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  src = fetchFromGitHub {
+    owner = "stumpylog";
+    repo = "gotenberg-client";
+    rev = "refs/tags/${version}";
+    hash = "sha256-xgkpVvklZrew+XOoqFKcbuDsTVfDda67R6YIxR3kzS8=";
+  };
+
+  propagatedBuildInputs = [
+    hatchling
```
Comment (natsukium) [inline pkgs/development/python-modules/gotenberg-client/default.nix:24]: `hatchling` is a build-time dependency, so it should go to `nativeBuildInputs`.

File: pkgs/development/python-modules/python-ipware/default.nix [hunk 6]
```diff
@@ -3,24 +3,30 @@
...
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/python-ipware/default.nix:11]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/python-ipware/default.nix (hunk 1 above)
Comment (leona-ya) [reply pkgs/development/python-modules/python-ipware/default.nix:10]: Paperless seems to be the only tool using it. Also nixpkgs-review runs fine

**Review by erikarvstedt — APPROVED** [review]
- [x] `nixosTests.paperless` succeeds on `x86_64-linux`

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
[v2.0.1](https://github.com/paperless-ngx/paperless-ngx/releases/tag/v2.0.1) has just been released.

**Review by erikarvstedt — APPROVED** [review]
- [x] `nixosTests.paperless` succeeds on `x86_64-linux`

**Review by lukegb — APPROVED** [review]
LGTM, will wait for ofborg to complete then merge, barring any objections in the meantime.

**Comment by lukegb** [comment]
@ofborg eval

**Pull request merged**